app/models/product.rb

```diff
@@ -85,4 +90,12 @@ def self.taken_product_titles
   private def words_in_permalink_separated_by_hypen
     permalink.split('-').length
   end
+
+  private def increment_parent_category_products_count
```

Comment:
seems repetitive, let's think about this

Reply:
Need to discuss about this